lib/typhoeus/response/header.rb

```diff
@@ -53,16 +53,20 @@ def process_line(header)
       # @return [ void ]
       def process_pair(key, value)
         set_value(key, value, self)
-        set_value(key.downcase, value, @sanitized)
+        @sanitized[key.downcase] = self[key]
       end
 
       # Sets value for key in specified hash
       #
       # @return [ void ]
       def set_value(key, value, hash)
-        if hash.has_key?(key)
-          hash[key] = [hash[key]] unless hash[key].is_a? Array
-          hash[key].push(value)
+        current_value = hash[key]
```

Comment:
This is the only thing I am not sure about. The check with `key?` is different than checking the value. It doesn't behave the same with `nil` and `false`.